ebu_tt_live/bindings/__init__.py

```diff
@@ -825,6 +843,75 @@ def content_to_string(self, begin=None, end=None):
 
 raw.br_type._SetSupersedingClass(br_type)
 
+class style_attr(IDMixin, RegionedElementMixin, LiveStyledElementMixin,
+               SemanticValidationMixin, raw.source_style, raw.element_style)
+    def _source_style(self, dataset):
+        source_style = style_type(
+            id=self.id
+        )
+        return source_style
+
+    def __source_style_copy__(self):
+         source_style = style_type(
+            id=self.id
+         )
+         return source_style
+
+    def _element_styling(self, dataset):
+        element_style = styling_type(
+            style=self._semantic_deconflicted_ids(attr_name='style', dataset=dataset)
+        )
+        return element_style
+
+    def __element_styling_copy__(self):
+        element_style = styling_type(
+            style=self.style
+        )
+        return element_style
+
+    def _semantic_before_traversal(self, dataset, parent_binding=None):
+        self._semantic_collect_applicable_styles(
+            dataset=dataset, style_type=style_type, parent_binding=parent_binding, defer_font_size=True
+        )
+        self._semantic_push_styles(dataset=dataset)
+
+    def _semantic_after_traversal(self, dataset, element_content=None, parent_binding=None):
+        self._semantic_postprocess_timing(dataset=dataset, element_content=element_content)
+
+    def _semantic_before_copy(self, dataset, element_content=None):
+        self._assert_in_segment(dataset=dataset, element_content=element_content)
+
+    def is_empty(self):
+        if len(self.style):
+            return False
+
+        if len(self.span):
+            return False
+
+        if len(self.region):
+            return False
+
+        if len(self.div):
+            return False
+
+        return True
+
+    def is_same(self):
```

Comment:
don't really think this method is appropriate,

its a bit overly complicated

`return source_style.id == element_style.style`

is sufficent